src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -1,9 +1,296 @@
-from middlewared.schema import accepts
-from middlewared.service import ConfigService
+from middlewared.schema import accepts, Bool, Dict, Str
+from middlewared.service import ConfigService, private
+from middlewared.utils import Popen, run
+
+import asyncio
+import os
+import shutil
+
+SYSDATASET_PATH = '/var/db/system'
 
 
 class SystemDatasetService(ConfigService):
 
     @accepts()
     async def config(self):
-        return await self.middleware.call('datastore.config', 'system.systemdataset', {'prefix': 'sys_'})
+        return await self.middleware.call('datastore.config', 'system.systemdataset', {'prefix': 'sys_', 'extend': 'systemdataset.config_extend'})
+
+    @private
+    async def config_extend(self, config):
+
+        # Add `is_decrypted` dynamic attribute
+        if config['pool'] == 'freenas-boot':
+            config['is_decrypted'] = True
+        else:
+            pool = await self.middleware.call('pool.query', [('name', '=', config['pool'])])
```

Comment:
Maybe we should add `query_first` and `query_one` methods like these:
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.first
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.one

Reply:
We kind of already have the "one"
self.middleware.call('pool.query', [('name', '=', config['pool'])], {'get': True})

However it raises an exception if not found. Perhaps it should be changed to return None